`ingest.py` cleanups, doc nits

ryan-williams opened the pull request:
**Issue and/or context:** while working on #2252, cleaned up a few things while trying to understand/audit them.

ryan-williams commented on apis/python/src/tiledbsoma/io/conversions.py:
```diff
@@ -46,7 +46,7 @@ def to_tiledb_supported_array_type(name: str, x: _MT) -> _MT:
     """
     if isinstance(x, (np.ndarray, sp.spmatrix)) or not isinstance(
-        x, pd.CategoricalDtype
+        x.dtype, pd.CategoricalDtype
```
I think this `if` condition was trivially evaluating as `True`, previously ([last commit](https://github.com/single-cell-data/TileDB-SOMA/commit/616823f4697c08db3043319cc68642551aaa44f5#diff-14e9163b10b9e187b5e590445e41f83c891a991bc670b9bda9cf194217532fc4L49-R50)).

ryan-williams commented on apis/python/src/tiledbsoma/io/ingest.py:
```diff
@@ -615,136 +573,60 @@ def from_anndata(
-                                _maybe_set(
-                                    obsm, key, arr, use_relative_uri=use_relative_uri
-                                )
-                            arr.close()
-                    measurement.obsm.close()
```
These `close`s don't exist in the `varm`/`obsp`/`varp` blocks below; they seem unnecessary, but let's confirm or refute that.

ryan-williams commented on apis/python/src/tiledbsoma/io/ingest.py:
```diff
@@ -615,136 +573,60 @@ def from_anndata(
-                    measurement.obsm.close()
-
-                if len(anndata.varm.keys()) > 0:  # do not create an empty collection
-                    _util.uri_joinpath(measurement_uri, "varm")
```
no-op?

ryan-williams commented on apis/python/src/tiledbsoma/io/ingest.py:
```diff
@@ -615,136 +573,60 @@ def from_anndata(
-                                )
-
-                if len(anndata.obsp.keys()) > 0:  # do not create an empty collection
-                    _util.uri_joinpath(measurement_uri, "obsp")
```
no-op?

ryan-williams commented on apis/python/src/tiledbsoma/io/ingest.py:
```diff
@@ -615,136 +573,60 @@ def from_anndata(
-                                )
-
-                if len(anndata.varp.keys()) > 0:  # do not create an empty collection
-                    _util.uri_joinpath(measurement_uri, "obsp")
```
no-op + copy-pasta (should be `"varp"`)

ryan-williams commented on apis/python/src/tiledbsoma/io/ingest.py:
```diff
@@ -615,136 +573,60 @@ def from_anndata(
+                ) -> None:
+                    ad_val = getattr(anndata, ad_key)
+                    if len(ad_val.keys()) > 0:  # do not create an empty collection
+                        val_uri = _util.uri_joinpath(measurement_uri, ad_key)
```
double-check: `measurement_uri` vs. `measurement.uri` here

johnkerl replied:
@ryan-williams for example
https://github.com/single-cell-data/TileDB-SOMA/blob/1.8.0/apis/python/src/tiledbsoma/io/ingest.py#L510-L517

ryan-williams reviewed (commented):
Left a couple comments highlighting areas we should look at, @johnkerl (once I get CI passing)

ryan-williams commented:
I've rebased this, and removed an `IngestionParams.__init__` refactor (https://github.com/single-cell-data/TileDB-SOMA/commit/39fe37ab) that @johnkerl and I were not quite sold on.
